pkg/gatewayserver/gatewayserver.go

```diff
@@ -636,6 +643,83 @@ func (gs *GatewayServer) handleUpstream(conn connectionEntry) {
 	}
 }
 
+func (gs *GatewayServer) handleLocationUpdates(conn connectionEntry) {
+	ctx := conn.Context()
+	uid := unique.ID(ctx, conn.Gateway())
+
+	logger := log.FromContext(ctx).WithFields(log.Fields(
+		"protocol", conn.Frontend().Protocol(),
+		"gateway_uid", uid,
+	))
+
+	var err error
+	var callOpt grpc.CallOption
+	callOpt, err = rpcmetadata.WithForwardedAuth(ctx, gs.AllowInsecureForCredentials())
+	if errors.IsUnauthenticated(err) {
+		callOpt = gs.WithClusterAuth()
+	} else if err != nil {
+		logger.WithError(err).Errorf("Not authenticated for location updates")
```

Comment:
Errors are intended for operations to look at, which is not bad user config. Also no need for the `f`. Same below

Reply:
Removed log message